webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala

```diff
@@ -225,66 +221,34 @@ case class TriplestoreServiceLive(
-      } else {
-        NonEmptyChunk
-          .fromIterableOption(rdfDataObjects)
-          .fold[Task[NonEmptyChunk[RdfDataObject]]](
-            ZIO.fail(BadRequestException("Cannot insert list with empty data into triplestore.")),
-          )(
-            ZIO.succeed(_),
-          )
-      }
-
+  ): Task[Unit] =
     for {
-      _    <- ZIO.logDebug("==>> Loading Data Start")
-      list <- calculateCompleteRdfDataObjectList
-      request <-
-        ZIO.foreach(list)(elem =>
-          for {
-            graphName <-
-              if (elem.name.toLowerCase == "default") {
-                ZIO.fail(TriplestoreUnsupportedFeatureException("Requests to the default graph are not supported"))
-              } else {
-                ZIO.succeed(elem.name)
-              }
-
-            uriBuilder <-
-              ZIO.attempt {
-                val uriBuilder: URIBuilder = new URIBuilder(paths.data)
-                uriBuilder.addParameter("graph", graphName) // Note: addParameter encodes the graphName URL
-                uriBuilder
-              }
-
-            rdfContents <- loadRdfObject(elem.path)
-
-            httpPost <-
-              ZIO.attemptBlocking {
-                val httpPost          = new HttpPost(uriBuilder.build())
-                val turtleContentType = ContentType.create(mimeTypeTextTurtle, "UTF-8")
-                httpPost.setEntity(new StringEntity(rdfContents, turtleContentType))
-                httpPost
-              }
-            responseHandler <- ZIO.attempt(returnInsertGraphDataResponse(graphName)(_))
-          } yield (httpPost, responseHandler),
-        )
-      _ <- ZIO.foreachDiscard(request)(elem => doHttpRequest(request = elem._1, processResponse = elem._2))
-      _ <- ZIO.logDebug("==>> Loading Data End")
+      _      <- ZIO.logDebug("==>> Loading Data Start")
+      objects = DefaultRdfData.data.toList.filter(_ => prependDefaults) ++ rdfDataObjects
+      _      <- ZIO.foreach(objects)(insertObjectIntoTriplestore)
+      _      <- ZIO.logDebug("==>> Loading Data End")
     } yield ()
-  }
 
   /**
-   * Load the RdfDataObject into some AbstractHttpEntity from either a relative ../$path or through a resource.
+   * Insert the RdfDataObject into the triplestore from either a relative ../$path or through a resource.
    */
-  private def loadRdfObject(path: String): Task[String] = {
-    val relativeFileEntity   = ZIO.readFile(Paths.get("..", path))
-    val resourceStringEntity = ZIO.attemptBlocking(Source.fromResource(path).mkString)
-
-    relativeFileEntity.orElse(resourceStringEntity)
-  }
+  private def insertObjectIntoTriplestore(rdfDataObject: RdfDataObject): Task[Unit] =
+    for {
+      graphName <- ZIO
+                     .succeed(rdfDataObject.name)
+                     .filterOrFail(_.toLowerCase != "default")(
+                       TriplestoreUnsupportedFeatureException("Requests to the default graph are not supported"),
+                     )
```

Comment:
Would this work?

```
      graphName <- ZIO
                     .fail(TriplestoreUnsupportedFeatureException("Requests to the default graph are not supported"))
                     .when(rdfDataObject.name.toLowerCase != "default")
                     .as(rdfDataObject.name)
```

The point is that although I'm not changing the value, it "goes" through a check before it can be used. Does this work for you?

Reply:
absolutely! (The initial version would have worked for me too 🙂 )